x/ibc/02-client/keeper/keeper.go

```diff
@@ -98,6 +98,67 @@ func (k Keeper) SetClientConsensusState(ctx sdk.Context, clientID string, height
 	store.Set(ibctypes.KeyConsensusState(height), bz)
 }
 
+// IterateConsensusStates provides an iterator over all stored consensus states.
+// objects. For each State object, cb will be called. If the cb returns true,
+// the iterator will close and stop.
+func (k Keeper) IterateConsensusStates(ctx sdk.Context, cb func(clientID string, cs exported.ConsensusState) bool) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, ibctypes.KeyClientStorePrefix)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		keySplit := strings.Split(string(iterator.Key()), "/")
+		// consensus key is in the format "clients/<clientID>/consensusState/<height>"
+		if len(keySplit) != 4 || keySplit[2] != "consensusState" {
```

Comment:
Why not check also `tokens[0] != "clients"`?

Reply:
we can, but it's redundant since we are already using it as the prefix for the iterator